rubicon/objc/objc.py

```diff
@@ -1324,6 +1325,10 @@ def __new__(cls, object_ptr):
         class_ptr = c_void_p(objc.object_getClass(object_ptr))
         objc_instance.__dict__['objc_class'] = ObjCClass(class_ptr)
 
+        # If it's an array, we need the internal counter in the instance to implement iterable.
+        if objc.class_getName(class_ptr).decode('utf-8').startswith('__NSArray'):
```

Comment:
What about `NSMutableArray`? Does it not belong to same category? And how about `NSDictionary`